kubeform/pkg/provider/provider.go

```diff
@@ -166,7 +161,24 @@ func createKubeconfig(d *schema.ResourceData, meta interface{}) error {
 }
 
 func deleteKubeconfig(d *schema.ResourceData, meta interface{}) error {
+	cfg, err := kubeconfigInit(d, meta)
+	if err != nil {
+		return err
+	}
+
+	po := clientcmd.NewDefaultPathOptions()
+	if path, ok := d.GetOk("path"); ok {
+		po.LoadingRules.ExplicitPath = path.(string)
+	}
+
+	// Retry until modifyConfig succeeds or times out.
+	log.Printf("[DEBUG] updating kubeconfig")
+	if !poll(cfg.configPollInterval, cfg.ConfigPollTimeout, updateConfig(po, cfg.kubeConfig, removeConfig)) {
+		return fmt.Errorf("cluster components never turned healthy")
```

Comment:
I am not entierely sure about why health check is needed here, if we are to delete it. One of the reason for deleting something may well be that it's broken.